stubdefaulter.py

```diff
@@ -85,6 +89,23 @@ def infer_value_for_default(
                     operator=libcst.Minus(),
                     expression=libcst.Integer(value=str(-param.default)),
                 )
+        elif type(param.default) is float:
+            if self.annotation_is_bool(node.annotation):
+                # Skip cases where the type is annotated as bool but the default is a float.
+                return None
+            if str(param.default) in {"nan", "inf", "-inf"}:
+                # Edge cases that it's probably not worth handling
+                return None
+            # `-0.0 == +0.0`, but we want to keep the sign,
+            # so use the string representation rather than the value itself
+            # to determine whether or not it's a negative float
+            if str(param.default).startswith("-"):
```

Comment:
I feel like there's got to be a better way for this, but I don't see anything in the float model. Maybe `math.copysign(param.default, -0.0) is param.default` but that feels even hackier (and haven't tested if it works).

Reply:
Yeah, I trawled through a few StackOverflow Q&As while preparing this, and couldn't really find anything better

Reply:
Actually this doesn't seem too bad:
```
In [25]: math.copysign(1, -0.0)
Out[25]: -1.0

In [26]: math.copysign(1, 0.0)
Out[26]: 1.0
```

```suggestion
            if math.copysign(1, param.default) < 0:
```

Reply:
Hmm, maybe we disagree on how hacky my solution using `str()` was. This sort-of feels like needless indirection to me.

(But no strong opinion; happy to keep what you've pushed. It's your project, after all :)

Reply:
I guess I have a stronger negative opinion about inspecting the str() result; I feel like it's relying on implementation details. For example, what if there's a subclass of float that overrides the `str()` behavior?

Reply:
Well, we know we haven't got a `float` subclass because we used `if type(param.default) is float` rather than `isinstance` :)

I guess the CPython repr for `float` itself could change, but that would be quite the breaking change.

Reply:
I do see your point now, though. Testing the repr might be safe and reliable, but we'd still be testing a "side effect" of the fact that the number had a negative sign, rather than testing the sign itself. So in that way, we'd be testing an implementation detail. Even if it's an implementation detail that we could safely count on never changing, it still wouldn't be very principled.

I'm persuaded; thanks!